src/lib.rs

```diff
@@ -135,8 +199,38 @@ from_impl!(String => String);
 into_impl!(&'a str);
 into_impl!(String);
 
+#[cfg(iter_cmp)]
+impl<T: AsRef<str>> PartialOrd for UniCase<T> {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[cfg(iter_cmp)]
+impl<T: AsRef<str>> Ord for UniCase<T> {
+    #[inline]
+    fn cmp(&self, other: &Self) -> Ordering {
+        match (&self.0, &other.0) {
+            (&Encoding::Ascii(ref x), &Encoding::Ascii(ref y)) => x.cmp(y),
+            (&Encoding::Unicode(ref x), &Encoding::Unicode(ref y)) => x.cmp(y),
+            (&Encoding::Ascii(ref x), &Encoding::Unicode(ref y)) => Unicode(x.as_ref()).cmp(&Unicode(y.0.as_ref())),
+            (&Encoding::Unicode(ref x), &Encoding::Ascii(ref y)) => Unicode(x.0.as_ref()).cmp(&Unicode(y.as_ref())),
+        }
```

Comment:
I think I'm misunderstanding, but my suggestion would be implemented for both the `Ord` and `PartialOrd` for `Unicode`? Either way, definitely not a blocker for this, I can experiment in future after this lands for my own curiosity.